plugins/catalog-backend/src/service/filterQuery.ts

```diff
@@ -70,3 +73,44 @@ export function translateFilterQueryEntryToEntityFilters(
 
   return filters;
 }
+
+type FieldMapper = (entity: Entity) => Entity;
+
+export function translateQueryToFieldMapper(
+  query: Record<string, any>,
+): FieldMapper {
+  if (!query.fields) {
+    return x => x;
+  }
+
+  const fieldsStrings = [query.fields].flat() as string[];
+
+  if (fieldsStrings.some(s => typeof s !== 'string')) {
+    throw new InputError(
+      'Only string type fields query parameters are supported',
+    );
+  }
+
+  const fields = fieldsStrings
+    .map(s => s.split(','))
+    .flat()
+    .map(s => s.trim())
+    .filter(Boolean);
+
+  if (!fields.length) {
+    return x => x;
+  }
+
+  return input => {
+    const output: RecursivePartial<Entity> = {};
+
+    for (const field of fields) {
+      const value = lodash.get(input, field);
```

Comment:
Would love to support something `jq` inspired like `relations[].type` for arrays, but could add that later. It's just wonky af to be able to use e.g. `relations[0].type`. Maybe we could just disallow the `[<n>]` syntax for now?

Thinking a few steps ahead I wanna be able to do something like a `expand=relations.type=owner&fields=relations[].entity.{kind,metadata.{name,namespace}}` type thing to expand relation entities. Not that exact syntax, but you get my point.

Reply:
Yep agreed, I forbade array based filtering now, and added tests